### ISSUE TITLE
Can basic-auth middleware support Fastly Compute@Edge?

### ISSUE THREAD

**Issue by yusukebe** [yusukebe/hono#47]
Most features of Hono work on Fastly Compute@Edge. Here is example:
https://github.com/yusukebe/hono/tree/master/example/compute-at-edge

That's great Hono supports both Cloudflare Workers and Compute@Edge. But, basic-auth middleware does not work on Compute@Edge because it does not provides `crypto`.

![SS](https://user-images.githubusercontent.com/10682/149447029-74e0319b-49b5-4bd1-a9e6-54aceecee94f.png)

The code I tried is here:

```js
app.use('/auth', Middleware.basicAuth({ username: 'user', password: 'pass' }))
```

Can basic-auth middleware support Fastly Compute@Edge? We can do it with `polyfill`? Using ` digest.hash_sha256` in C@E? If it is difficult to support C@E without dependencies, let us give up. Because, **Hono must not depends any modules**.

If anyone knows, let me know.

**Comment by metrue** [comment]
Compute@Edge does not have built-in `crypto`,  currently the **basis-auth** middleware is using [crypto.subtle.digest](https://github.com/yusukebe/hono/blob/master/src/util.ts#L65) to build the `timingSafeEqual` to do the constant time comparison, as you said, polyfill the `digest` with [digest.hash_sha256](https://developer.fastly.com/reference/vcl/functions/cryptographic/digest-hash-sha256/) should work. 

Another thing we may need to take care of it is the `Buffer` which is not ready in WASM either but we use it in https://github.com/yusukebe/hono/blob/master/src/middleware/basic-auth/basic-auth.ts#L36, the document gives an example polyfill with webpack, but I think it's not what we need because **hono** should not depend on user's webpack configure.

```javascript
module.exports = { // ...
  // Shimming globals
  plugins: [
    new webpack.ProvidePlugin({
        // Polyfill the URL standard
        URL: "core-js/web/url",
        // Polyfill Node.js' buffer (requires https://www.npmjs.com/package/buffer)
        Buffer: ["buffer", "Buffer"],
    }),
  ],
```

**Comment by yusukebe** [comment]
Fixed on #86
